ydlidar_ros2_driver_node:
  ros__parameters:
    port: /dev/ydlidar       # Vamos usar o alias, é mais seguro que ttyUSB0
    frame_id: laser_frame
    ignore_array: ""
    baudrate: 115200         # <-- Tente 115200 primeiro com as novas flags abaixo
    lidar_type: 1
    device_type: 0
    sample_rate: 5           # <-- Baixe para 5 (padrão mais seguro) em vez de 9
    abnormal_check_count: 4
    fixed_resolution: true
    reversion: true          # <-- Mude para true
    inverted: true           # <-- Mude para true
    auto_reconnect: true
    isSingleChannel: true    # <-- IMPORTANTE: Mude para true
    intensity: false
    support_motor_dtr: true  # <-- IMPORTANTE: Mude para true para ligar o motor
    angle_max: 180.0
    angle_min: -180.0
    range_max: 64.0
    range_min: 0.1
    frequency: 6.0
    invalid_range_is_inf: false
    debug: false
